activerecord/lib/active_record/persistence.rb

```diff
@@ -179,6 +179,7 @@ def becomes(klass)
       became.instance_variable_set("@new_record", new_record?)
       became.instance_variable_set("@destroyed", destroyed?)
       became.instance_variable_set("@errors", errors)
+      became.errors.instance_variable_set("@base", became)
```

Comment:
> Closing due to inactivity. If you're still interested in working on this, let me know. For this to be mergeable, it needs to not mutate the original errors, but instead copy them. The implementation should add an internal method for this, too, not reach into the internals of another object.

@sgrif should be something like:

```
became.errors.merge(errors)

instead of 

became.instance_variable_set("@errors", errors)
became.errors.instance_variable_set("@base", became)
```

?